### PR TITLE
blobwars: init at 2.00

### PR DESCRIPTION
**Pull request by iblech**
###### Motivation for this change

Blobwars is an old and neat platform game available in Debian and Arch. This pull request adds blobwars to nixpkgs.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/blobwars/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{ stdenv, fetchurl, pkgconfig, gettext, SDL2, SDL2_image, SDL2_mixer, SDL2_net, SDL2_ttf, zlib }:
+
+stdenv.mkDerivation rec {
+  pname = "blobwars";
+  version = "2.00";
+  src = fetchurl {
```
Comment (SuperSandro2000) [inline pkgs/games/blobwars/default.nix:6]:
```suggestion
  version = "2.00";

  src = fetchurl {
```

File: pkgs/games/blobwars/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    url = "mirror://sourceforge/${pname}/${pname}-${version}.tar.gz";
+    sha256 = "c406279f6cdf2aed3c6edb8d8be16efeda0217494acd525f39ee2bd3e77e4a99";
+  };
+
+  nativeBuildInputs = [ pkgconfig gettext ];
```
Comment (SuperSandro2000) [inline pkgs/games/blobwars/default.nix:11]:
```suggestion
  nativeBuildInputs = [ pkg-config gettext ];
```

File: pkgs/games/blobwars/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{ stdenv, fetchurl, pkgconfig, gettext, SDL2, SDL2_image, SDL2_mixer, SDL2_net, SDL2_ttf, zlib }:
```
Comment (SuperSandro2000) [inline pkgs/games/blobwars/default.nix:1]:
```suggestion
{ stdenv, fetchurl, pkg-config, gettext, SDL2, SDL2_image, SDL2_mixer, SDL2_net, SDL2_ttf, zlib }:
```

File: pkgs/games/blobwars/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    mkdir $out/bin
+    mv $out/games/blobwars $out/bin
+    rmdir $out/games
+    cp -r {data,gfx,sound,music} $out/share/games/blobwars/
+    # fix world readable bit
+    find $out/share/games/blobwars/. -type d -exec chmod 755 {} +
+    find $out/share/games/blobwars/. -type f -exec chmod 644 {} +
+  '';
+
+  meta = with stdenv.lib; {
```
Comment (sikmir) [inline pkgs/games/blobwars/default.nix:27]: What about meta.homepage?

File: pkgs/games/blobwars/default.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+  nativeBuildInputs = [ pkgconfig gettext ];
+  buildInputs = [ SDL2 SDL2_image SDL2_mixer SDL2_net SDL2_ttf zlib ];
+  NIX_CFLAGS_COMPILE = [ "-Wno-error" ];
+
+  makeFlags = [ "PREFIX=$(out)" "RELEASE=1" ];
+
+  postInstall = ''
+    mkdir $out/bin
+    mv $out/games/blobwars $out/bin
```
Comment (sikmir) [inline pkgs/games/blobwars/default.nix:19]:
```suggestion
    install -Dm755 $out/games/blobwars -t $out/bin
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108884` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>blobwars</li>
</ul>
</details>

File: pkgs/games/blobwars/default.nix (hunk 4 above)
Comment (iblech) [reply pkgs/games/blobwars/default.nix:28]: Oops, good catch, will fix this in the force-push!

File: pkgs/games/blobwars/default.nix (hunk 2 above)
Comment (iblech) [reply pkgs/games/blobwars/default.nix]: Thanks, I didn't realize that `pkgconfig` was an older alias. Fixed.

File: pkgs/games/blobwars/default.nix [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  buildInputs = [ SDL2 SDL2_image SDL2_mixer SDL2_net SDL2_ttf zlib ];
+  NIX_CFLAGS_COMPILE = [ "-Wno-error" ];
+
+  makeFlags = [ "PREFIX=$(out)" "RELEASE=1" ];
+
+  postInstall = ''
+    mkdir $out/bin
+    install -Dm755 $out/games/blobwars -t $out/bin
+    rm $out/games/blobwars
+    rmdir $out/games
```
Comment (SuperSandro2000) [inline pkgs/games/blobwars/default.nix:22]:
```suggestion
    rm -r $out/games/
```

Comment (iblech) [reply pkgs/games/blobwars/default.nix:22]: Good idea!

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108884` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>blobwars</li>
</ul>
</details>

**Comment by iblech** [comment]
I just force-pushed to resolve the merge conflict.

File: pkgs/games/blobwars/default.nix [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    mkdir $out/bin
+    install -Dm755 $out/games/blobwars -t $out/bin
+    rm -r $out/games
+    cp -r {data,gfx,sound,music} $out/share/games/blobwars/
+    # fix world readable bit
+    find $out/share/games/blobwars/. -type d -exec chmod 755 {} +
+    find $out/share/games/blobwars/. -type f -exec chmod 644 {} +
+  '';
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/games/blobwars/default.nix:28]:
```suggestion
  meta = with lib; {
```

File: pkgs/games/blobwars/default.nix [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+  };
+
+  nativeBuildInputs = [ pkg-config gettext ];
+  buildInputs = [ SDL2 SDL2_image SDL2_mixer SDL2_net SDL2_ttf zlib ];
+  NIX_CFLAGS_COMPILE = [ "-Wno-error" ];
+
+  makeFlags = [ "PREFIX=$(out)" "RELEASE=1" ];
+
+  postInstall = ''
+    mkdir $out/bin
```
Comment (SuperSandro2000) [inline pkgs/games/blobwars/default.nix:19]:
```suggestion
```

**Comment by iblech** [comment]
Thank you for the new review, @SuperSandro2000! :-)

**Pull request merged**